New gems & pickaxe

nekomaster1000 opened the pull request:
Builds off of #40, so be sure to approve that before this one!

nekomaster1000 committed:
Removed Supra & Ultima Shovels and Hoes, amongst others

Also removed
- Kinetic Opal Sword and Pickaxe
- Power Pyrite Pickaxe
- Spirit Garnet Axe
Accordingly adjusted the recipes and advancements of the Ultima Axe, too

nekomaster1000 committed:
Added Divine Beryl, Spider Kunzite and Unbreaking Iolite gems

- Added them to the gems tags, the English language file, and The Collector advancement
- To add: Gem Blocks, Ore Blocks, Loot Tables, Ore Spawning

nekomaster1000 committed:
Added Divine Beryl, Spider Kunzite and Unbreaking Iolite blocks

- Added Block, Ore and Deepslate blocks for the three new gems.
- Added aforementioned blocks to appropriate advancements and tags.
- Added generation for the new ore blocks, too.
Divine Beryl natively spawns under swamps and dark forests,
Spider Kunzite natively spawns under forests,
and Unbreaking Iolite natively spawns under mushroom isles.

nekomaster1000 committed:
Added Unbreaking Iolite Pickaxe (& Updated Sprites)

- Including recipes and crafting advancements
- Added Unbreaking Iolite to the crafting recipe of the Ultima Pickaxe - Using the Sword Catalyst recipe, Peridots are replaced with a Peridot Block to fit everything in, meaning the Peridot Block had to be added to the gems tag
- Added new gems to the animated tab icon
- Fixed a small texture mistake on the Paragon Boots
- Fixed the Haste Peridot Pickaxe using the wrong Ingredient.of item
- Gave Power Pyrite Sword & Axe Sharpness VI

Hellion-Games commented on src/main/java/org/infernalstudios/miningmaster/config/MiningMasterConfig.java:
```diff
@@ -78,6 +81,21 @@ public MiningMasterConfig(ForgeConfigSpec.Builder builder) {
+        divineBerylEnabled = builder
+                .comment("Determines if Divine Beryl will generate.")
+                .translation(MiningMaster.MOD_ID + ".config.tooltip.divineBerylEnabled")
+                .define("diveAquamarineEnabled", true);
```
Didn't change copy paste

Hellion-Games commented on src/main/java/org/infernalstudios/miningmaster/config/MiningMasterConfig.java:
```diff
@@ -78,6 +81,21 @@ public MiningMasterConfig(ForgeConfigSpec.Builder builder) {
+        spiderKunziteEnabled = builder
+                .comment("Determines if Spider Kunzite will generate.")
+                .translation(MiningMaster.MOD_ID + ".config.tooltip.spiderKunziteEnabled")
+                .define("diveAquamarineEnabled", true);
```
Didn't change copy paste

Hellion-Games commented on src/main/java/org/infernalstudios/miningmaster/config/MiningMasterConfig.java:
```diff
@@ -78,6 +81,21 @@ public MiningMasterConfig(ForgeConfigSpec.Builder builder) {
+        unbreakingIoliteEnabled = builder
+                .comment("Determines if Unbreaking Iolite will generate.")
+                .translation(MiningMaster.MOD_ID + ".config.tooltip.unbreakingIoliteEnabled")
+                .define("diveAquamarineEnabled", true);
```
Didn't change copy paste

Hellion-Games commented on src/main/java/org/infernalstudios/miningmaster/enchantments/FloatationEnchantment.java:
```diff
@@ -77,7 +77,7 @@ public void doPostAttack(LivingEntity user, Entity target, int level) {
         if (target instanceof LivingEntity) {
             LivingEntity livingTarget = (LivingEntity) target;
-            livingTarget.addEffect(new MobEffectInstance(MobEffects.LEVITATION, 20 * level));
+            livingTarget.addEffect(new MobEffectInstance(MobEffects.LEVITATION, 4 * level));
```
Why is this being reduced down to 1/5 of a second? That seems extremely low

nekomaster1000 replied:
Explained over Discord, but for prosperity: Levitation completely disables a ground-bound melee opponent's ability to attack. 1 second of Levitation max is more than enough time to get consecutive bow or arrow strikes in on an opponent.

Hellion-Games commented on src/main/java/org/infernalstudios/miningmaster/init/MMBlocks.java:
```diff
@@ -62,6 +69,9 @@ public class MMBlocks {
     public static final RegistryObject<Block> HASTE_PERIDOT_BLOCK = registerBlockWithDefaultItem("haste_peridot_block", () -> new Block(getProperties(Blocks.DIAMOND_BLOCK)));
     public static final RegistryObject<Block> LUCKY_CITRINE_BLOCK = registerBlockWithDefaultItem("lucky_citrine_block", () -> new Block(getProperties(Blocks.DIAMOND_BLOCK)));
     public static final RegistryObject<Block> DIVE_AQUAMARINE_BLOCK = registerBlockWithDefaultItem("dive_aquamarine_block", () -> new Block(getProperties(Blocks.DIAMOND_BLOCK)));
+    public static final RegistryObject<Block> DIVINE_BERYL_BLOCK = registerBlockWithDefaultItem("divine_beryl_block", () -> new GemOreBlock(getProperties(Blocks.DIAMOND_BLOCK)));
+    public static final RegistryObject<Block> SPIDER_KUNZITE_BLOCK = registerBlockWithDefaultItem("spider_kunzite_block", () -> new GemOreBlock(getProperties(Blocks.DIAMOND_BLOCK)));
+    public static final RegistryObject<Block> UNBREAKING_IOLITE_BLOCK = registerBlockWithDefaultItem("unbreaking_iolite_block", () -> new GemOreBlock(getProperties(Blocks.DIAMOND_BLOCK)));
```
These should be Blocks, not GemOreBlocks

Hellion-Games commented on src/main/resources/data/miningmaster/advancements/miningmaster/geo_dude.json:
```diff
@@ -91,13 +91,37 @@
     },
-    "miningmaster:kinetic_opal_block": {
+    "miningmaster:divine_beryl_block": {
       "trigger": "minecraft:inventory_changed",
       "conditions": {
         "items": [
           {
             "items": [
-              "miningmaster:kinetic_opal_block"
+              "miningmaster:divine_beryl_block"
+            ]
+          }
+        ]
+      }
+    },
```
If I'm reading this correctly, you removed the Kinetic Opal Block from this advancement, why? I don't see it added anywhere else in this file so it seems to be gone altogether.

nekomaster1000 replied:
Oh whoops! It looks like in copy-pasting for the other gems, I accidentally glazed over and replaced the last remaining KO entry on this one. My bad!

Hellion-Games commented on src/main/resources/data/miningmaster/advancements/miningmaster/the_collector.json:
```diff
@@ -91,6 +91,42 @@
         ]
       }
     },
+    "miningmaster:divine_beryl": {
+      "trigger": "minecraft:inventory_changed",
+      "conditions": {
+        "items": [
+          {
+            "items": [
+              "miningmaster:divine_beryl"
+            ]
+          }
+        ]
+      }
+    },
+    "miningmaster:spider_kunzite": {
+      "trigger": "minecraft:inventory_changed",
+      "conditions": {
+        "items": [
+          {
+            "items": [
+              "miningmaster:spider_kunzite"
+            ]
+          }
+        ]
+      }
+    },
+    "miningmaster:unbreaking_iolite": {
+      "trigger": "minecraft:inventory_changed",
+      "conditions": {
+        "items": [
+          {
+            "items": [
+              "miningmaster:unbreaking_iolite"
+            ]
+          }
+        ]
+      }
+    },
```
You added these, but didn't actually add them to the requirements for the advancement.

Hellion-Games commented on src/main/resources/data/miningmaster/advancements/recipes/forging/ultima_axe_4.json:
```diff
@@ -2,29 +2,17 @@
   "rewards": {
     "recipes": [
-      "miningmaster:forging/ultima_axe_4"
+      "miningmaster:forging/ultima_axe_5"
```
Why did you change this to 5 in this file, the 4 file, but delete the 5 file?

nekomaster1000 replied:
That's... A good question! I don't know! I deleted the 5 file due to the loss of the Garnet Axe, I think, but I don't know why I made this odd mistake here.

Hellion-Games commented on src/main/resources/data/miningmaster/advancements/recipes/forging/ultima_pickaxe_1.json:
```diff
@@ -30,25 +30,37 @@
     },
-    "has_kinetic_opal": {
+    "has_haste_peridot_block": {
       "trigger": "minecraft:inventory_changed",
       "conditions": {
         "items": [
           {
             "items": [
-              "miningmaster:kinetic_opal"
+              "miningmaster:haste_peridot_block"
+            ]
+          }
+        ]
+      }
+    },
```
Why is this now checking for a whole block of Haste Peridot? We may have talked about this, but I forget.

Hellion-Games commented on src/main/resources/data/miningmaster/recipes/forging/ultima_pickaxe_1.json:
```diff
@@ -11,16 +11,19 @@
     },
     {
-      "item": "miningmaster:haste_peridot"
+      "item": "miningmaster:haste_peridot_block"
```
Why does this one take a block, recipe 2 takes a single gem, and it looks like recipe 3 doesn't take any haste peridot?

nekomaster1000 replied:
Recipe 2 takes three Peridots (which is still a mistake) whilst recipe 3 doesn't take any Peridots because the catalyst is a Peridot Pickaxe. I realise the block solution is a little clunky, so I've decided to rejig them again without the block.

Dariensg reviewed (changes requested):
Few things:
1. You didn't change the catalyst tag at all for the changed tools.
2. I don't think you changed the ultima recipes to take the new tools instead of/in addition to the old ones.
3. The ultima recipes as a whole look a little wrong to me. I pointed out some of the issues, but I think they should be gone through carefully to make sure everything is right.

Everything else I noticed should be specified in comments on the relevant sections. Hopefully I didn't miss anything!

nekomaster1000 committed:
Added missing config translation keys + Review changes

- Changed the Ultima Pickaxe recipes to no longer require a block of Peridot, just individual gems (and extensively the necessary advancements and gem tag)
- Added the Unbreaking Iolite Pickaxe to the list of catalysts and gave it its own ultima pick recipe (Previously left out intentionally, but decided to add it in for consistency)
- Edited the catalysts tag, removing the removed items
- Removed the shovel and hoe related advancements :(

Dariensg reviewed (approved):
LGTM

nekomaster1000 committed:
Merge branch '1.18' into new-gems
